plugins/utils/ccc.nix

```diff
@@ -5,50 +5,218 @@
+          "ccc.picker.css_rgb"
+          "ccc.picker.css_hsl"
+          "ccc.picker.css_hwb"
+          "ccc.picker.css_lab"
+          "ccc.picker.css_lch"
+          "ccc.picker.css_oklab"
+          "ccc.picker.css_oklch"
+        ];
+        example = [
+          "ccc.picker.hex"
+          "ccc.picker.css_rgb"
+          "ccc.picker.css_hsl"
+          "ccc.picker.css_hwb"
+          "ccc.picker.css_lab"
+          "ccc.picker.css_lch"
+          "ccc.picker.css_oklab"
+          "ccc.picker.css_oklch"
+          "ccc.picker.css_name"
+        ];
+        description = "List of formats that can be detected by `:CccPick` to be activated.";
+        apply = mapToRawLua;
+      };
+
+      highlight_mode =
+        helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "bg"
+          "fg"
+          "background"
+          "foreground"
+        ]
+        ''
+          Option to highlight text foreground or background.
+          It is used to `output_line` and `highlighter`.
+        '';
+
       lsp = helpers.defaultNullOpts.mkBool true ''
-        If true, highlight using LSP. If a language server with the color
-        provider is not attached to a buffer, it falls back to highlight with
-        pickers. See also `:help ccc-option-lsp`.
+        Whether to enable LSP support.
+        The color information is updated in the background and the result is used by `:CccPick` and
+        highlighter.
       '';
+
+      highlighter = {
+        auto_enable = helpers.defaultNullOpts.mkBool false ''
+          Whether to enable automatically on `BufEnter`.
+        '';
+
+        filetypes = helpers.defaultNullOpts.mkListOf types.str "[]" ''
+          File types for which highlighting is enabled.
+          It is only used for automatic highlighting by `ccc-option-highlighter-auto-enable`, and is
+          ignored for manual activation.
+          An empty table means all file types.
+        '';
+
+        excludes = helpers.defaultNullOpts.mkListOf types.str "[]" ''
+          Used only when `ccc-option-highlighter-filetypes` is empty table.
+          You can specify file types to be excludes.
+        '';
+
+        lsp = helpers.defaultNullOpts.mkBool true ''
+          If true, highlight using LSP.
+          If language server with the color provider is not attached to a buffer, it falls back to
+          highlight with pickers.
+          See also `:help ccc-option-lsp`.
+        '';
+
+        update_insert = helpers.defaultNullOpts.mkBool true ''
+          If true, highlights will be updated during insert mode.
+          If false, highlights will not be updated during editing in insert mode, but will be
+          updated on `InsertLeave`.
+        '';
+      };
+
+      convert = mkOption {
+        type = listOfRawLua;
+        # TODO
+        default = [
+          "{ ccc.picker.hex, ccc.output.css_rgb }"
+          "{ ccc.picker.css_rgb, ccc.output.css_hsl }"
+          "{ ccc.picker.css_hsl, ccc.output.hex }"
```

Comment:
I think this option should be a list of raw lua pairs, or at least a list of list of raw lua

Reply:
It could also be an attr set, as this represents nicely pairings (where the keys & values would be raw lua)

Reply:
The latter would not be RFC-42 style compliant, right ?
I went with the `listOf (listOf strLua)` approach.